server/routes/api/landing_page.py

```diff
@@ -317,28 +321,60 @@ def scale_series(numerator, denominator):
 
     The date of the two time series may not be exactly aligned. Here we use
     year alignment to match two date. If no denominator is found for a
-    numerator, then the data is removed.
+    numerator, then the data is removed. We allow the denominator to be
+    MAX_DENOMINATOR_BACK_YEAR stale.
     """
     data = {}
+
+    num_date_fmt = None
+    denom_date_fmt = None
+    for date in denominator.keys():
+        denom_date_fmt = get_date_fmt(date)
+    for date in numerator.keys():
+        num_date_fmt = get_date_fmt(date)
+    if not denom_date_fmt or not num_date_fmt:
+        return {}
+
+    latest_denom_date = max(denominator.keys())
+    latest_denom_date_obj = datetime.datetime.strptime(latest_denom_date,
+                                                       denom_date_fmt)
+
     for date, value in numerator.items():
         if date in denominator:
             if denominator[date] > 0:
                 data[date] = value / denominator[date]
-            else:
-                data[date] = 0
-        else:
-            try:
-                numerator_year = get_year(date)
-                for i in range(0, MAX_DENOMINATOR_BACK_YEAR + 1):
-                    year = str(numerator_year - i)
-                    if year in denominator:
-                        if denominator[year] > 0:
-                            data[date] = value / denominator[year]
-                        else:
-                            data[date] = 0
-                        break
-            except ValueError:
-                return {}
+                continue
+        # Try various fuzzy matchings
+        try:
+            date_obj = datetime.datetime.strptime(date, num_date_fmt)
+        except ValueError:
+            continue
+
+        # Case 1: if denominator is stale...
+        if latest_denom_date_obj < date_obj:
+            # Use latest date if within reason.
+            if (date_obj - latest_denom_date_obj <=
+                    datetime.timedelta(days=365 * MAX_DENOMINATOR_BACK_YEAR)):
+                data[date] = value / denominator[latest_denom_date]
+            # No other dates will work. Go to next date regardless.
+            continue
+
+        if denom_date_fmt == '%Y':
+            for i in range(0, MAX_DENOMINATOR_BACK_YEAR + 1):
+                year = str(date_obj.year - i)
+                if year in denominator:
+                    if denominator[year] > 0:
+                        data[date] = value / denominator[year]
+                    break
+        elif denom_date_fmt == '%Y-%m':
+            for i in range(0, 12 * MAX_DENOMINATOR_BACK_YEAR + 1):
+                year_month = datetime.datetime(date_obj.year, date_obj.month,
```

Comment:
not sure how efficient these operations are. may be worth doing something quick and dirty with some string manipulation and abs math